### PR TITLE
LeadTwo chevron

### PR DESCRIPTION
**Pull request by arne182**
Draw also second car indicator from LeadTwo Data

### CODE REVIEW

**Comment by arne182** [comment]
@pd0wm 
do you really want 
  if ((scene->lead_status2) &&(fabs(scene->lead_d_rel - scene->lead_d_rel2)>0.1)||(fabs(scene->lead_y_rel - scene->lead_y_rel2)>0.1))) {
as condition?

File: selfdrive/ui/ui.cc [hunk 1]
```diff
@@ -357,11 +357,17 @@ void handle_message(UIState *s, Message * msg) {
     struct cereal_RadarState datad;
     cereal_read_RadarState(&datad, eventd.radarState);
     struct cereal_RadarState_LeadData leaddatad;
+    struct cereal_RadarState_LeadData leaddatae;
     cereal_read_RadarState_LeadData(&leaddatad, datad.leadOne);
+    cereal_read_RadarState_LeadData(&leaddatae, datad.leadTwo);
```
Comment (pd0wm) [inline selfdrive/ui/ui.cc]: Can you reuse the same struct? Just do the `cereal_read_RadarState_LeadData(&leaddatae, datad.leadTwo);` after you're done with the first lead.

File: selfdrive/ui/paint.cc [hunk 2]
```diff
@@ -404,6 +404,21 @@ static void ui_draw_world(UIState *s) {
     draw_chevron(s, scene->lead_d_rel+2.7, scene->lead_y_rel, 25,
                   nvgRGBA(201, 34, 49, fillAlpha), nvgRGBA(218, 202, 37, 255));
   }
+  if (scene->lead_status2) {
+    // Draw lead2 car indicator
```
Comment (pd0wm) [inline selfdrive/ui/paint.cc]: Can you refactor this duplicated code into a `draw_lead(scene, d_rel, v_rel, y_rel)` function?

**Review by pd0wm — CHANGES_REQUESTED** [review]
Yes. That condition sounds fine. Otherwise the leads are not distinct.

File: selfdrive/ui/ui.cc (hunk 1 above)
Comment (arne182) [reply selfdrive/ui/ui.cc]: Done. Even though you have a spelling mistake in your comment

File: selfdrive/ui/paint.cc (hunk 2 above)
Comment (arne182) [reply selfdrive/ui/paint.cc]: let us see if the tests come back with a tick

File: selfdrive/ui/paint.cc [hunk 3]
```diff
@@ -108,6 +108,23 @@ static void draw_chevron(UIState *s, float x_in, float y_in, float sz,
   nvgRestore(s->vg);
 }
 
+static void draw_lead(UIState *s, float d_rel, float v_rel, float y_rel){
+    const UIScene *scene = &s->scene;
```
Comment (pd0wm) [inline selfdrive/ui/paint.cc]: scene is unused in this function.

**Comment by HaraldSchafer** [comment]
> @pd0wm
> do you really want
> if ((scene->lead_status2) &&(fabs(scene->lead_d_rel - scene->lead_d_rel2)>0.1)||(fabs(scene->lead_y_rel - scene->lead_y_rel2)>0.1))) {
> as condition?

There needs to be some check that they are different otherwise might be a lot of almost overlapping chevrons. Could also just do distance maybe (fabs(scene->lead_d_rel - scene->lead_d_rel2)>3)

File: selfdrive/ui/paint.cc (hunk 3 above)
Comment (arne182) [reply selfdrive/ui/paint.cc]: didn't know if this is needed. The other functions had it. Thanks for the explaination

**Review by pd0wm — APPROVED** [review]

**Pull request merged**